src/js/cli_engine.js

```diff
@@ -0,0 +1,423 @@
+
+                const outString = this.#GUI.textarea.val();
+                this.#history.add(outString);
+                this.executeCommands(outString);
+                this.#GUI.textarea.val('');
+            }
+        });
+
+        this.#GUI.textarea.keyup(event => {
+            if (this.#cliAutoComplete && this.#cliAutoComplete.isOpen()) {
+                return; // disable history keys if autocomplete is open
+            }
+
+            if (event.key == "Up") {
+                this.#GUI.textarea.val(this.#history.prev());
+            }
+
+            if (event.key == "Down") {
+                this.#GUI.textarea.val(this.#history.next());
+            }
+        });
+
+        // give input element user focus
+        this.#GUI.textarea.focus();
+    }
+
+    // Close attempts to send an `exit` to the flight controller. Notably, it uses the `_lineWithBuffer` function as there could be pending CLI buffer data.
+    close(callback) {
+        let line = 'exit\r';
+        if (this.#cliBufferContainsPartialCommand) {
+            line = this.#lineWithBuffer(line);
+        }
+        this.#send(line, function () {
+            // we could handle this "nicely", but this will do for now
+            // (another approach is however much more complicated):
+            // we can setup an interval asking for data lets say every 200ms,
+            // when data arrives, callback will be triggered and tab switched
+            // we could probably implement this someday
+            callback?.();
+            CONFIGURATOR.cliEngineActive = false;
+            CONFIGURATOR.cliEngineValid = false;
+            CONFIGURATOR.cliTab = '';
+        });
+    }
+
+    // executeCommands splits (on \n) the given output string into an array of strings and returns a promise that resolves after all of the commands have been sent.
+    async executeCommands(commandString) {
+        const commandArray = commandString.split("\n");
+        this.executeCommandsArray(commandArray);
+    }
+
+    // executeCommandsArray
+    async executeCommandsArray(commandsArray) {
+        this.#reportSendCommandsProgress(0);
+        for (let i = 0, delay = 0; i < commandsArray.length; i++) {
+            let line = commandsArray[i].trim();
+            if (line.length === 0) continue;
+
+            if (delay) await new Promise(resolve => setTimeout(resolve, delay));
+
+            if (this.#cliBufferContainsPartialCommand) {
+                line = this.#lineWithBuffer(line);
+            }
+
+            await new Promise(resolve => this.sendLine(line, resolve));
+
+            this.#reportSendCommandsProgress(100.0 * i / commandsArray.length);
+
+            delay = line.toLowerCase().startsWith('profile') ? this.#profileSwitchDelayMs : 0;
+            delay = line.toLowerCase().startsWith('rateprofile') ? this.#rateProfileSwitchDelayMs : 0;
+            
+            if (delay === 0) {
+                delay = this.#lineDelayMs;
+            }
```

Comment:
Make this an if chain. As it is, `profile` delay does not work.

Reply:
true, will fix